src/main/flight/interpolated_setpoint.c

```diff
@@ -58,9 +69,42 @@ FAST_CODE_NOINLINE float interpolatedSpApply(int axis, bool newRcFrame, ffInterp
         const float rxInterval = currentRxRefreshRate * 1e-6f;
         const float rxRate = 1.0f / rxInterval;
 
-        const float setpointSpeed = (rawSetpoint - prevRawSetpoint[axis]) * rxRate;
-        const float setpointAcceleration = (setpointSpeed - prevSetpointSpeed[axis]) * pidGetDT();
+        float setpointSpeed = (rawSetpoint - prevRawSetpoint[axis]) * rxRate;
+        float setpointAcceleration = setpointSpeed - prevSetpointSpeed[axis];
+        const float holdSteps = 2.0f;
+
+        // Glitch reduction code for identical packets
+        if (setpointSpeed == 0 && fabsf(rawSetpoint) < 0.98f * ffMaxRate[axis]) {
+            // identical packets, not at full deflection
+            if (holdCount[axis] < holdSteps && fabsf(rawSetpoint) > 2.0f && !bigStep[axis]) {
+                // holding the entire previous speed is best for missed packets, but bad for early packets
+                setpointSpeed = prevSetpointSpeed[axis] + prevAcceleration[axis];
+                setpointAcceleration = prevAcceleration[axis];
+                holdCount[axis] += 1.0f;
+            // hit timeout, or at centre and not moving, or prev big step, lock speed and acceleration to zero
+            } else {
+                holdCount[axis] -= 1.0f;
```

Comment:
Is this correct?  holdSteps is currently 2, which allows us to 'replace' one repeated frame with a 'guess'...

But what about frames going 1,2,3,4,4,4,4

The first 4 is good, holdCount=1.
The 2nd 4 is repeated frame, holdCount is 1 and 1<2 so we use previous speed + acceleration and set holdCount=2.
The 3rd 4 is again repeated..  holdCount=2 so we hit this else block and line 86 sets holdCount-=1, so its 1 again.
The 4th 4 is again repeated.. holdCount=1 again, so now we can get back into line 80 if block?  That doesn't seem like we want that?

Is line 86 needed?  If deleted I think things might be fine, or maybe its even better to do holdCount[axis] = holdSteps, so if somehow we get in this else block we 'disable' any holding going forward until we get a non-repeated frame?

Reply:
Oh I remember other issue with doing holdCount[axis] -= 1...  this is an unsigned int.  What happens if you start getting repeated frames but sticks are centered?  Won't you will keep getting into this else block over and over again long as sticks stay centered right?  And each time you get into it holdCount-=1... only will have to repeat a couple frames before holdCount=0 and you do holdCount-=1 and it becomes INT_MAX.

Reply:
It could be that this can be improved.  We need a way to know, when the sticks move again, what the status was before they moved. 
In the example you gave, 1,2,3,4,4,4,4,5,5 (adding 5's when sticks start), we only want normal acceleration and boost.

Reply:
Maybe line 86 needs to be `holdCount[axis] = 3.0f;`
Then after the 3,4 transition, the first 4,4 will set holdCount to 2, and do the right things, then the second 4,4 will set holdCount to 3, and from then on while setpoint speed is zero, holdCount will stay 3. 
When we start moving, only if holdCount is 2 (meaning one identical packet, ie the first 4,4) will we reduce the FF and boost.  If holdCount was 3, ie flat, line 98 will set it back to 1 and be in normal detection state.
I think this will solve the problem?

Reply:
set to above solution, tested, works well.  thanks